libsolidity/ast/AST.cpp

```diff
@@ -596,6 +601,12 @@ set<VariableDeclaration::Location> VariableDeclaration::allowedDataLocations() c
 		return set<Location>{ Location::Unspecified };
 }
 
+u256 VariableDeclaration::externalIdentifier() const
+{
+	solAssert(isStateVariable(), "Can only be called for state variables");
```

Comment:
@chriseth Should I assert the variable `isPublic()` here as well?

Reply:
yes, probably better to do that.